torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -324,9 +325,41 @@ void ConvertGraphToONNXProto(
   }
 }
 
+// this function checks wheather the blocks of If node have the same return
+// type.
+bool IsBlockReturnType(Node* n) {
+  TORCH_INTERNAL_ASSERT(n->kind() == ::c10::onnx::If);
+  auto then_block = n->blocks()[0];
+  auto else_block = n->blocks()[1];
+  for (size_t i = 0; i < n->outputs().size(); i++) {
+    // check the type
+    auto then_block_type = then_block->outputs()[i]->type();
```

Comment:
How about checking shape difference from else and then?

Reply:
In this PR we are only considering the issue of Type propagation, since it is affecting one of the tests.
I will add possible issue of shape propagation in my document.